src/main/scala/com/netaporter/precanned/HttpServerMock.scala

```diff
@@ -50,8 +52,22 @@ class HttpServerMock extends Actor {
 
   var responses = Vector.empty[ExpectAndRespondWith]
 
-  private def responseFor(request: HttpRequest) =
-    responses.find(_.expects(request)).map(_.respondWith)
+  private def responseFor(request: HttpRequest) = {
```

Comment:
Could you please extract side-effect of marking expectation fired to a separate method to keep it all clear?
It could be two separate methods like `findExpectationFor` (without side effects) and `markExpectationFired` (with side effect).